### ISSUE TITLE
obspy warns of duplicate resource identifiers; jing doesn't

### ISSUE THREAD

**Issue by nackerley** [obspy/obspy#2002]
The attached event in QuakeML format [20090115.0928.xml.zip](https://github.com/obspy/obspy/files/1471740/20090115.0928.xml.zip) validates fine in jing, but if you:
```python
from obspy import read_events
catalog = read_events('20090115.0928.xml')
catalog.write('20090115.0928_rewritten.xml', format='QuakeML', validate=True)
```

The result is four warnings like this when reading:
```
C:\Users\nackerle\AppData\Local\Continuum\anaconda3\envs\py3\lib\site-packages\obspy\core\event\base.py:378: UserWarning: The resource identifier 'smi:ca.seismo/stn_mag/solid/arrid/20090115.0928001/20090115.0928003' already exists and points to another object: 'StationMagnitude(resource_id=ResourceIdentifier(id="smi:ca.seismo/stn_mag/solid/arrid/20090115.0928001/20090115.0928003"), origin_id=ResourceIdentifier(id="smi:ca.seismo/origin/solid/20090115.0928001"), mag=-0.0, station_magnitude_type='ML', amplitude_id=ResourceIdentifier(id="smi:ca.seismo/amplitude/solid/arrid/20090115.0928001/20090115.0928003"))'.It will now point to the object referred to by the new resource identifier.
  self.resource_id.set_referred_object(self)
```
And an error when writing:
```
AssertionError: The final QuakeML file did not pass validation.
```

I expect QuakeML-1.2.rng to give the same validation result for the same QuakeML in obspy and jing. If obspy additionally checks for duplicate resource identifiers while writing, that's ok, but this should raise warnings, not errors.

**Comment by krischer** [comment]
I think you are using `jing` incorrectly or maybe you have a different version?

```bash
$ java -jar ~/Downloads/jing-20091111/bin/jing.jar ~/Downloads/QuakeML-1.2.rng 20090115.0928.xml
/Users/lion/temp/obspy_weird_validation/20090115.0928.xml:173:15: error: element "mag" incomplete; missing required element "value"
```

If you fix this, ObsPy also correctly validates the file.

**Comment by krischer** [comment]
The duplicate resource id was a bug in ObsPy due to how we internally handled these identifiers but that should be fixed in the latest version.

**Comment by nackerley** [comment]
It seems you're saying the duplicate resource id warnings were a red herring; this .xml does have duplicate ids, and as such should generate warnings, I would think.

As for the validation with jing, I'm not sure how I ended up with the wrong .rng, but once I re-downloaded from http://quake.ethz.ch/schema/rng/ I was able to see the results you did, specifically the mal-formed mag element at line 173.

Here's a minimum working example to get that mal-formed magnitude:
``` python
from obspy.core.event import Event,Magnitude,Catalog
from obspy import read_events

event = Event()
event.magnitudes = [Magnitude()]

catalog = Catalog()
catalog.events = [event]

catalog.write('test.xml', format='QuakeML', validate=True)  # fails
catalog.write('test.xml', format='QuakeML')  # works
catalog2 = read_events('test.xml')  # works
```

Fyi, this came about because a station_magnitude_contribution which were all given zero weights. It's a weird situation but not totally unreasonable.

**Comment by krischer** [comment]
> It seems you're saying the duplicate resource id warnings were a red herring; this .xml does have duplicate ids, and as such should generate warnings, I would think.

Yes and no. ObsPy's job during reading any file is not to check its validity but only to read it to its best ability and only raise if something is really really wrong. I guess we could debate reintroducing the warning when a resource identifier appears multiple times - please open a separate issue if you feel this should be the case.

The latest ObsPy version is able to keep track of different resources even with the same identifier (see https://github.com/obspy/obspy/pull/1644).
